src/plugins/howler.convolver.js

```diff
@@ -0,0 +1,354 @@
+       | convolverSend |              |         
+       -----------------              |       
+                     ^                |          
+                     |                |       
+                     |                |       
+---------------    -----------        |       
+|              |-->| _fxSend |        |
+|              |   -----------        |
+| bufferSource |         _________    |
+|              | ----->  | _node | ----              
+---------------          ---------            
+
+        */  
+
+
+/*!
+ *  Convolver Plugin - Adds support for convolving howls with built-in or custom impulse responses
+ *                     where Web Audio is supported. Convolution is most commonly used to apply 
+ *                     reverb characteristics of a space to an arbitrary sound, but creative effects can
+ *                     also be achieved by convolving your sounds with other audio files. 
+ *                     (http://iub.edu/~emusic/etext/synthesis/chapter4_convolution.shtml)
+ *                     - Plugin by Jack Campbell
+ *  
+ *  howler.js v2.0.4
+ *  howlerjs.com
+ *
+ *  (c) 2013-2017, James Simpson of GoldFire Studios
+ *  goldfirestudios.com
+ * 
+ *
+ *  MIT License
+ */
+
+(function() {
+    
+      'use strict';
+      
+      /** Global Methods **/
+      /***************************************************************************/
+    
+      /**
+       * Load an impulse response and register its name to be used as a convolver
+       * @param  {String} convolverName Name of convolver to connect to
+       * @param  {String} impulseResponse URL of impulse response audio file to load
+       * @param  {Function} callback Callback called when impulse response is loaded
+       * @return {HowlerGlobal}
+       */
+      HowlerGlobal.prototype.addConvolver = function(convolverName, impulseResponse, callback) {
+        var self = this;
+    
+        // Stop right here if not using Web Audio.
+        if (!self.ctx || !self.ctx.listener) {
+          return self;
+        }
+        if(!self._convolvers) { self._convolvers = {}; }
+        // search if convolver already exists by that name
+        if(self._convolvers[convolverName])
+        {
+            console.warn('A convolver already exists under this name.');
+            return self;
+        }
+
+
+        var xhr = new XMLHttpRequest();
+        if (!impulseResponse) {
+            console.log("Could not find IR at supplied path");
+            return;
+        }
+
+        xhr.open("GET", impulseResponse, true);
+        xhr.responseType = "arraybuffer";
+        xhr.onreadystatechange = function() {
+            if (xhr.readyState === 4) {
+                if (xhr.status < 300 && xhr.status > 199 || xhr.status === 302) {
+                    Howler.ctx.decodeAudioData(xhr.response, function(buffer) {
+                        // create convolver
+                        let convolver = Howler.ctx.createConvolver();
```

Comment:
I don't think applying a convolver node to the entire Howler object is ever desirable (it'll affect all Howl instances). At the very least it should be applied only to a single Howl instance.